src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/ParameterValidationAnalysis/ParameterValidationAnalysis.ParameterValidationDataFlowOperationVisitor.cs

```diff
@@ -319,7 +319,7 @@ private void ProcessRegularInvocationOrCreation(IMethodSymbol targetMethod, Immu
                             var notValidatedLocations = GetNotValidatedLocations(argument);
                             foreach (var location in notValidatedLocations)
                             {
-                                var parameter = (IParameterSymbol)location.SymbolOpt;
+                                var parameter = (IParameterSymbol)location.SymbolOpt!;
```

Comment:
💭 `ContainsKey` should work fine even with `null`

```suggestion
                                var parameter = (IParameterSymbol?)location.SymbolOpt;
```

Reply:
This generates subsequent nullability false positives. Additionally, symbol is guaranteed to be non-null for locations created for parameters, so the suppression is essentially an assertion.